curvefs/src/client/mds_client.cpp

```diff
@@ -17,37 +17,226 @@
 
 /*
  * Project: curve
- * Created Date: Thur May 27 2021
- * Author: xuchaojie
+ * Created Date: Thur Jun 15 2021
+ * Author: lixiaocui
  */
 
 #include "curvefs/src/client/mds_client.h"
 
 namespace curvefs {
 namespace client {
 
-CURVEFS_ERROR MdsClientImpl::Init(const MdsOption& mdsOpt) {
-    return CURVEFS_ERROR::OK; }
+CURVEFS_ERROR MdsClientImpl::MDSRPCExcutor::DoRPCTask(RPCFunc task) {
+    brpc::Channel channel;
+    std::string mdsaddr = opt_.mdsaddr;
+    int ret = channel.Init(mdsaddr.c_str(), nullptr);
+    if (ret != 0) {
+        LOG(WARNING) << "Init channel failed, addr = " << mdsaddr;
+        return static_cast<CURVEFS_ERROR>(-EHOSTDOWN);
+    }
+
+    brpc::Controller cntl;
+    cntl.set_timeout_ms(opt_.rpcTimeoutMs);
+
+    ret = task(&channel, &cntl);
+    return static_cast<CURVEFS_ERROR>(ret);
+}
+
+
+CURVEFS_ERROR
+MdsClientImpl::Init(const MdsOption &mdsOpt, MDSBaseClient *baseclient) {
+    excutor_.SetOption(mdsOpt);
+    mdsbasecli_ = baseclient;
+    return CURVEFS_ERROR::OK;
+}
+
+#define RPCTask [&](brpc::Channel * channel, brpc::Controller * cntl) -> int // NOLINT
```

Comment:
"using RpcTask = [&](brpc::Channel * channel, brpc::Controller * cntl) -> int;" is  better.